main/vstring.c

```diff
@@ -81,12 +81,12 @@ extern vString *vStringNew (void)
 
 extern void vStringPut (vString *const string, const int c)
```

Comment:
> @masatake So you think we still need the inline keyword? As I mentioned from my experience the compiler will ignore it in any case.

Are you using compiler other than gcc?

```
$ objdump -d ./ctags | grep '\(callq.*vStringPut\)\|>:'
...
000000000044ec30 <plistFindTagsUnderKey>:
  44ed88:	e8 c3 cf fc ff       	callq  41bd50 <vStringPut>
...
```
`vStringPut` is still `call`ed. It is not inlined. gcc may do so only if a function is `static`. 
 "Linker and Loader"  says optimizing corss-compile-units can be done with linker. GNU ld may  do it if we give more option flags. https://gcc.gnu.org/wiki/LinkTimeOptimization

Don't make a PR for introducing LTO. Instead think about rewriting ctags with javascript or go:-P